lib/bigquery/job.js

```diff
@@ -156,35 +188,9 @@ nodeutil.inherits(Job, ServiceObject);
  * @example
  * job.cancel(function(err, apiResponse) {
  *   // Check to see if the job completes successfully.
- *   onJobComplete(function(err) {
- *     if (!err) {
- *       // Job cancelled successfully.
- *     }
- *   });
+ *   job.on('error', function(err) {});
+ *   job.on('complete', function(metadata) {});
```

Comment:
I think the usage patten of mixing callbacks with emitters just throws me off. What if instead of a callback we emitted a `running` (or similar) event

```js
job.cancel()
  .on('error', console.log)
  .on('running', function() {apiResponse}) // job has started (optional)
  .on('complete', function() {metadata}); // job is complete
  
```